rework_pysatl_mpest/core/mixture.py

```diff
@@ -244,15 +268,15 @@ def pdf(self, X: ArrayLike) -> NDArray[float64]:
 
         Returns
         -------
-        NDArray[np.float64]
+        NDArray[DType]
             The PDF values corresponding to each point in :attr:`X`.
         """
 
-        X = np.asarray(X, dtype=float64)
+        X = np.asarray(X, dtype=self.dtype)
         component_pdfs = np.array([comp.pdf(X) for comp in self.components])
         return np.asarray(np.dot(self.weights, component_pdfs))
```

Comment:
I would rewrite this method for better numerical stability:
```suggestion
        return np.exp(self.lpdf(X))
```

Then you will need to synchronize the corresponding test like this:

<img width="829" height="379" alt="изображение" src="https://github.com/user-attachments/assets/b7971953-db04-4ce1-8888-b959180d7082" />